File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/ApplepayInitRequest.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class ApplepayInitRequest : SignBaseRequest
{
    [JsonProperty("orderNo")] public string? OrderNo { get; set; }

    [JsonProperty("payMethod")] public string? PayMethod { get; set; }

    [JsonProperty("clientIp")] public string? ClientIp { get; set; }

    [JsonProperty("totalAmount")] public long? TotalAmount { get; set; }

    [JsonProperty("currency")] public string? Currency { get; set; }

    [JsonProperty("closePayment")] public bool? ClosePayment { get; set; }

    [JsonProperty("payload")] public string? Payload { get; set; }

    [JsonProperty("returnUrl")] public string? ReturnUrl { get; set; }

    [JsonProperty("returnMethod")] public string? ReturnMethod { get; set; }

    [JsonProperty("customer")] public Customer? Customer { get; set; }

    [JsonProperty("order")] public Order? Order { get; set; }

    [JsonProperty("sdkUsed")] public bool? SdkUsed { get; set; }

    [JsonProperty("merchantData")] public string? MerchantData { get; set; }

    [JsonProperty("ttlSec")] public long? TtlSec { get; set; }

    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, MerchantId);
        Add(sb, OrderNo);
        Add(sb, Dttm);
        Add(sb, PayMethod);
        Add(sb, ClientIp);
        Add(sb, TotalAmount);
        Add(sb, Currency);
        Add(sb, ClosePayment);
        Add(sb, Payload);
        Add(sb, ReturnUrl);
        Add(sb, ReturnMethod);
        Add(sb, Customer?.ToString());
        Add(sb, Order?.ToString());
        Add(sb, SdkUsed);
        Add(sb, MerchantData);
        Add(sb, TtlSec);
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/ApplepayInitRequest.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;
import cz.monetplus.mips.eapi.v19.connector.entity.ext.Extension;
import lombok.*;

@EqualsAndHashCode(callSuper = false)
@AllArgsConstructor @NoArgsConstructor @RequiredArgsConstructor
@Data
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class ApplepayInitRequest extends SignBase {

    @NonNull
    private String merchantId;
    @NonNull
    private String orderNo;
    private String payMethod;
    @NonNull
    private String clientIp;
    @NonNull
    private Long totalAmount;
    @NonNull
    private String currency;
    private Boolean closePayment;
    @NonNull
    private String payload;
    @NonNull
    private String returnUrl;
    private String returnMethod; //[GET, POST]
    private Customer customer;
    private Order order;
    private Boolean sdkUsed;
    private String merchantData;

    private Integer ttlSec;
    private List<Extension> extensions;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, merchantId);
        add(sb, orderNo);
        add(sb, dttm);
        add(sb, payMethod);
        add(sb, clientIp);
        add(sb, totalAmount);
        add(sb, currency);
        add(sb, closePayment);
        add(sb, payload);
        add(sb, returnUrl);
        add(sb, returnMethod);
        if (null != customer) add(sb, customer.toSign());
        if (null != order) add(sb, order.toSign());
        add(sb, sdkUsed);
        add(sb, merchantData);
        add(sb, ttlSec);
        deleteLast(sb);
        return sb.toString();
    }
}
